hub/ui/src/components/search-bar/SearchBar.tsx

```diff
@@ -36,8 +36,11 @@ const SearchBar: React.FC = (props: any) => {
   const onTextChanged = (e: any) => {
     const value = e;
     let suggestions: any = [];
-    const regex = new RegExp(`${value}`, 'i');
-    suggestions = props.TaskDataList.sort().filter((v: any) => regex.test(v.name));
+    const nameregex = /-/gi;
+    const regex = new RegExp(`${ value.replace(nameregex, ' ') }`, 'i');
+    suggestions = props.TaskDataList.sort().filter((v: any) =>
```

Comment:
please filter and then sort so that the number of records to be sorted is less.

Reply:
here sort doesn't need and can be remove because while dispatching suggestions sorting is happening already .